lib/Date.php

```diff
@@ -266,4 +266,44 @@ private function numberOfDaysInMonth(int $year, int $month): int
             \sprintf('%d-%02d-%02d', $year, $month, 1),
         )->format('t');
     }
+
+    /**
+     * @return self The earliest date from the provided dates.
+     */
+    public static function earliestOf(self ...$datesToCompare): self
```

Comment:
```suggestion
    public static function earliestOf(self $date, self ...$datesToCompare): self
```
nit: this will require at least one argument in the fn invocation.